src/helpers/chain.js

```diff
@@ -0,0 +1,582 @@
+ */
+export const getLatestSubmittedRound = async (oracle) => {
+  let fromBoard = makeFromBoard();
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(`:published.wallet.${oracle}`, leader, {
+    unserializer,
+  });
+
+  // Get offers
+  let offers = await getOffers(follower);
+
+  return Number(
+    offers[0]["status"]["invitationSpec"]["invitationArgs"][0]["roundId"]
+  );
+};
+
+/**
+ * Function to check if a price submission was successful for a specific round
+ * @param {string} oracle oracle address
+ * @param {number} feedOfferId the offer id of the feed to check for
+ * @param {number} roundId the round Id which is checked
+ * @returns {boolean} whether a submission was successful for a specific round
+ */
+export const checkSubmissionForRound = async (oracle, feedOfferId, roundId) => {
+  let fromBoard = makeFromBoard();
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(`:published.wallet.${oracle}`, leader, {
+    unserializer,
+  });
+
+  // Get offers
+  let offers = await getOffers(follower);
+
+  // Loop through offers starting from last offer
+  for (let i = 0; i < offers.length; i++) {
+
+    // Get current offer
+    let currentOffer = offers[i]
+
+    // If a price invitation and for the correct feed
+    let invitationType = currentOffer["status"]["invitationSpec"][
+    "invitationMakerName"];
+    let previousOffer = currentOffer["status"]["invitationSpec"][
+    "previousOffer"];
+    if (
+        invitationType === "PushPrice" &&
+        previousOffer === feedOfferId
+    ) {
+      let offerRound = Number(
+        currentOffer["status"]["invitationSpec"]["invitationArgs"][0]["roundId"]
+      );
+
+      /**
+       * If it is an offer for the round we are checking for and there is no
+       * error
+       */
+      if (
+        offerRound === roundId &&
+        !currentOffer["status"].hasOwnProperty("error")
+      ) {
+        return true;
+      } else if (
+        /**
+         * If the currentOffer has no error and the offerRound is less than the
+         * roundId, we return false since no new submissions for future rounds
+         * can be created before completing the current round
+         */
+        offerRound < roundId &&
+        !currentOffer["status"].hasOwnProperty("error")
+      ) {
+        return false;
+      }
+    }
+  }
+  return false;
+};
+
+/**
+ * Function to query price from chain
+ * @param {string} feed feed name of the price to query (Ex. ATOM-USD)
+ * @returns {number} the latest price
+ */
+export const queryPrice = async (feed) => {
+  try {
+    // Read value from vstorage
+    const capDataStr = await readVStorage(feed, false);
+
+    let capData;
+
+    try {
+      //parse the value
```

Comment:
```suggestion
      // Parse the value
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/29244ced3c6d34785c4643507480179d98d7d8c6